### PR TITLE
Improve formatting of generated config

### PR DESCRIPTION
**Pull request by CyberDem0n**
- order sections similar to sample configs
- add warnings and comments to `bootstrap.dcs` section.
- add `tags` and `log` sections.
- use discovered IPs in `postgresql.connect_address` and `postgresql.listen`
- set `wal_level` to `replica` for PostgreSQL 9.6+
- make unit tests pass with python 3.6
- improve config validator so it doesn't complain when some ints are strings in YAML file.

### CODE REVIEW

File: patroni/config_generator.py [hunk 1]
```diff
@@ -130,16 +136,82 @@ def get_template_config(cls) -> Dict[str, Any]:
...
+            if isinstance(self.config[section_name], dict):
+                yield ''
+            yield self._format_block({section_name: self.config[section_name]})
+
+    def _format_config(self) -> Iterator[str]:
+        """Format current :attr:`~AbstractConfigGenerator.config` and enrich it with some comments.
+
+        :yields: formatted lines or blocks that represent a text output of the YAML document.
+        """
+        for name in ('scope', 'namespace', 'name', 'restapi', 'ctl' 'citus',
```
Comment (hughcapet) [inline patroni/config_generator.py:174]: log section?

Comment (CyberDem0n) [reply patroni/config_generator.py:174]: BTW, should we add it do the generated config?

Comment (hughcapet) [reply patroni/config_generator.py:174]: definitely won't hurt

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**